### PR TITLE
introduce FactoryBot

### PR DESCRIPTION
**Pull request by mockdeep**
This adds test factories to make setting up records for the tests much
cleaner. I also tossed in the `Faker` gem for some fun test data.

http://www.rubydoc.info/gems/factory_bot/file/GETTING_STARTED.md
https://github.com/stympy/faker

### CODE REVIEW

**Comment by dladowitz** [comment]
Super great. This was on my list of things to do.

File: spec/factories/users.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :user do
+    display_name { Faker::DragonBall.character }
```
Comment (dladowitz) [inline spec/factories/users.rb]: I was thinking of publishing `Faker::Crypto.people` and  `Faker::Crypto.jargon`

Comment (dladowitz) [reply spec/factories/users.rb]: Someone already created a bitcoin address faker: https://github.com/stympy/faker/blob/master/doc/bitcoin.md

Comment (mockdeep) [reply spec/factories/users.rb]: I bet they'd accept a PR if you wanted to add it.

**Pull request merged**